src/main/java/org/jahia/modules/serversettings/flow/WebprojectHandler.java

```diff
@@ -1228,4 +1228,19 @@ public String getDefaultTemplateSetId() {
     public int getNumberOfSites() throws JahiaException {
         return sitesService.getNbSites() - 1;
     }
+
+    /**
+     * This method is calling the LicenseCheckerService to get the site limit value
+     * and check if the limit has been reached
+     *
+     * @return <code>true</code> if the limit is reached, <code>false</code> otherwise
+     */
+    public boolean isSiteLimitReached() throws JahiaException {
+        Optional<Long> value = LicenseCheckerService.Stub.getSiteLimit();
+        if (value.isPresent()) {
+            return (sitesService.getNbSites() - 1) >= value.get();
```

Comment:
The minus 1 part is quite confusing here. Maybe we should add a method in SiteService for counting the number of non system sites (impl would be the one from SiteCountValidator), and thus both this method and the SiteCountValidator would use the new method.

Reply:
Yep I think that can be done and it would be quite easy